File: dcp-solutions/2024-12-08/solution.php
```php
<?php
class TreeNode {
    public $val = 0;
    public $left = null;
    public $right = null;

    function __construct($val) {
        $this->val = $val;
    }
}

function isValidBST($root) {
    return helper($root, PHP_INT_MIN, PHP_INT_MAX);
}

function helper($node, $lower, $upper) {
    if (!$node) return true;

    if ($node->val <= $lower || $node->val >= $upper) return false;
    $left = helper($node->left, $lower, $node->val);
    $right = helper($node->right, $node->val, $upper);

    return $left && $right;
}
```

File: dcp-solutions/2024-12-08/solution.go
```markdown
package main

import (
    "errors"
    "math Big"
)

func isValidBST(root *TreeNode) bool {
    var val = math Big.-bigOne;
    return check(root, val);
}

func check(node *TreeNode, val Big) bool {
    if node == nil {
        return true;
    }

    if node.Val < val {
        return false;
    }
    if node.Val > val {
        return false;
    }

    left := check(node.Left, val Big.Min(float64(node.Val), float64(val)))
    if left == false{
        return false
    }
    right := check(node.Right, val Big.Max(float64(node.Val), float64(val)))
    if right == false{
        return false
    }
    return true;
}

type TreeNode struct {
    Val   int
    Left  *TreeNode
    Right *TreeNode
}
```